plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2932,17 +2936,26 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
                 .orElseThrow(() -> new IllegalStateException("Storage table missing in definition of materialized view " + materializedViewName));
 
         Table icebergTable = catalog.loadTable(session, storageTableName);
-        String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
+        Snapshot currentSnapshot = icebergTable.currentSnapshot();
+        String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnFunctions = Optional.ofNullable(currentSnapshot)
+                .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                .orElse("");
+
         if (dependsOnTables.isEmpty()) {
+            // If the source of the MV is a PTF then source tables will be empty but source functions won't
+            if (!dependsOnFunctions.isEmpty()) {
+                return new MaterializedViewFreshness(UNKNOWN, Optional.of(Instant.ofEpochMilli(currentSnapshot.timestampMillis())));
+            }
             // Information missing. While it's "unknown" whether storage is stale, we return "stale".
-            // Normally dependsOnTables may be missing only when there was no refresh yet.
+            // Normally dependsOnTables may be missing only when there was no refresh yet or when the source is a PTF
```

Comment:
or when MV has no sources (`CREATE M V foo AS VALUES 1`)

i think we should distinguish DEPENDS_ON_TABLES missing from DEPENDS_ON_TABLES being empty.
Missing means we don't have information.
Empty means we have the information and list of tables depended on happens to be empty.

Reply:
Should this be in scope of this PR ?

Reply:
I think it is quite complex to distinguish these to cases correctly:
- currently both DEPENDS_ON_TABLES and DEPENDS_ON_FUNCTIONS will be empty when REFRESH was not run, no matter source is used
- when REFRESH is run and there is no source table or no source function DEPENDS_ON_TABLES will contain "" and also DEPENDS_ON_FUNCTIONS will contain "" - to correctly distinguish lack of source table from lack of information this should be changed and i am afraid that will pull another changes .
So in a separate PR maybe ?